zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/GitNotebookRepo.java

```diff
@@ -61,28 +61,34 @@ public GitNotebookRepo(ZeppelinConfiguration conf) throws IOException {
       localRepo.create();
     }
     git = new Git(localRepo);
-    maybeAddAndCommit(".");
+    checkpoint(".", "initialization commit");
   }
 
   @Override
   public synchronized void save(Note note) throws IOException {
     super.save(note);
-    maybeAddAndCommit(note.getId());
   }
 
-  private void maybeAddAndCommit(String pattern) {
+  /* implemented as git add+commit
+   * @param pattern is the noteId
+   * @param commitMessage is a part of commit message (checkpoint name)
+   * (non-Javadoc)
+   * @see org.apache.zeppelin.notebook.repo.VFSNotebookRepo#checkpoint(String, String)
+   */
+  @Override
+  public void checkpoint(String pattern, String commitMessage) {
     try {
       List<DiffEntry> gitDiff = git.diff().call();
       if (!gitDiff.isEmpty()) {
         LOG.debug("Changes found for pattern '{}': {}", pattern, gitDiff);
         DirCache added = git.add().addFilepattern(pattern).call();
         LOG.debug("{} changes are about to be commited", added.getEntryCount());
-        git.commit().setMessage("Updated " + pattern).call();
+        git.commit().setMessage("Updated " + pattern + ": " + commitMessage).call();
```

Comment:
good point! Actually as I've checked now, `pattern` from `setMessage` isn't  used anywhere else. addressed it in b0fa21919e613784340b9ec9efbd63c47aae9b59